### PR TITLE
Handle Yaml date and datetime values.

### PR DESCRIPTION
**Pull request by Natim**
We might have to handle that at the kinto-http level.

```
kinto_wizard/yaml2kinto.py:166: in initialize_server
    logger.debug('Sending batch:\n\n%s' % batch.session.requests)
/usr/lib/python3.6/contextlib.py:88: in __exit__
    next(self.gen)
.tox/py36/lib/python3.6/site-packages/kinto_http/__init__.py:105: in batch
    batch_session.send()
.tox/py36/lib/python3.6/site-packages/kinto_http/batch.py:61: in send
    resp, headers = self.session.request(**kwargs)
.tox/py36/lib/python3.6/site-packages/kinto_http/session.py:94: in request
    resp = requests.request(method, actual_url, **kwargs)
.tox/py36/lib/python3.6/site-packages/requests/api.py:58: in request
    return session.request(method=method, url=url, **kwargs)
.tox/py36/lib/python3.6/site-packages/requests/sessions.py:498: in request
    prep = self.prepare_request(req)
.tox/py36/lib/python3.6/site-packages/requests/sessions.py:441: in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
.tox/py36/lib/python3.6/site-packages/requests/models.py:312: in prepare
    self.prepare_body(data, files, json)
.tox/py36/lib/python3.6/site-packages/requests/models.py:462: in prepare_body
    body = complexjson.dumps(json)
.tox/py36/lib/python3.6/site-packages/simplejson/__init__.py:382: in dumps
    return _default_encoder.encode(obj)
.tox/py36/lib/python3.6/site-packages/simplejson/encoder.py:296: in encode
    chunks = self.iterencode(o, _one_shot=True)
.tox/py36/lib/python3.6/site-packages/simplejson/encoder.py:378: in iterencode
    return _iterencode(o, 0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <simplejson.encoder.JSONEncoder object at 0x7fc9530032e8>, o = datetime.date(2018, 6, 22)

    def default(self, o):
        """Implement this method in a subclass such that it returns
            a serializable object for ``o``, or calls the base implementation
            (to raise a ``TypeError``).
    
            For example, to support arbitrary iterators, you could
            implement default like this::
    
                def default(self, o):
                    try:
                        iterable = iter(o)
                    except TypeError:
                        pass
                    else:
                        return list(iterable)
                    return JSONEncoder.default(self, o)
    
            """
        raise TypeError('Object of type %s is not JSON serializable' %
>                       o.__class__.__name__)
E       TypeError: Object of type date is not JSON serializable

.tox/py36/lib/python3.6/site-packages/simplejson/encoder.py:273: TypeError
```

### CODE REVIEW

File: kinto_wizard/kinto2yaml.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def _sorted_principals(permissions):
     """
     items = dct.items()
     results = await asyncio.gather(*(item[1] for item in items))
-    return dict(zip((item[0] for item in items), results))
+    return {k: v for k, v in zip((item[0] for item in items), results) if v is not None}
```
Comment (Natim) [inline kinto_wizard/kinto2yaml.py]: I don't remember why I changed the behaviour here :confused:

**Comment by Natim** [comment]
Coverage failures were fixed in #56

File: kinto_wizard/kinto2yaml.py (hunk 1 above)
Comment (Natim) [reply kinto_wizard/kinto2yaml.py]: Apparently it fixes the case of empty buckets with the `introspect_server` call:

```
        with open("tests/dumps/dump-toto.yaml") as f:
>           assert f.read() == generated
E           AssertionError: assert 'natim:\n  co...a5494b127c4\n' == 'date: null\nn...a5494b127c4\n'
E             + date: null
E               natim:
```

Comment (leplatrem) [reply kinto_wizard/kinto2yaml.py]: It could deserve some intermediary variables and/or comments :)

File: kinto_wizard/kinto2yaml.py [hunk 2]
```diff
@@ -54,13 +54,16 @@ def _sorted_principals(permissions):
...
+            result = {
+                'permissions': _sorted_principals(permissions),
+                'collections': {
+                    collection: await introspect_collection(client, bid, collection,
+                                                            data=data, records=records)
+                }
             }
-        }
+        except kinto_exceptions.CollectionNotFound:
+            return None
```
Comment (leplatrem) [inline kinto_wizard/kinto2yaml.py]: This seems unrelated to dates ;)

**Comment by Natim** [comment]
> You could rename the PR title...
I did rename it is prior to your review, isn't it good enough speaking about date and datetime usage?

File: kinto_wizard/kinto2yaml.py (hunk 2 above)
Comment (Natim) [reply kinto_wizard/kinto2yaml.py]: Shall I create another PR?

File: kinto_wizard/kinto2yaml.py (hunk 1 above)
Comment (leplatrem) [reply kinto_wizard/kinto2yaml.py]:
```suggestion
    keys_results =  zip((item[0] for item in items), results)
    return {k: v for k, v in keys_results if v is not None}
```
